### ISSUE TITLE
`ERC1155`: reorder of `KeyType`s in `_balances`

### ISSUE THREAD

**Issue by pcaversaccio** [OpenZeppelin/openzeppelin-contracts#5023]
It might be nit, but you define `_balances` in your [`ERC1155`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/11dc5e3809ebe07d5405fe524385cbe4f890a08b/contracts/token/ERC1155/ERC1155.sol#L23C72-L23C81) implementation accordingly:

```solidity
mapping(uint256 id => mapping(address account => uint256)) private _balances;
```

but the [`balanceOf`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/ERC1155.sol#L64-L66) function uses the (switched) EIP-1155-compliant way:

```solidity
function balanceOf(address account, uint256 id) public view virtual returns (uint256) {
    return _balances[id][account];
}
```

I would propose to change `_balances` to:

```solidity
mapping(address account => mapping(uint256 id => uint256)) private _balances;
```

in order to avoid confusion. My guess, why it has been implemented like this in OZ, is the official EIP implementation [here](https://github.com/enjin/erc-1155/blob/master/contracts/ERC1155.sol#L16).

**Comment by ernestognw** [comment]
Hey @pcaversaccio,

Took a quick look at the code because I agree it feels weird. So far I agree with the rationale but changing the order of the mapping is a breaking change for upgradeable contracts. I don't think we can fix that until 6.0 and it doesn't harm.

**Comment by pcaversaccio** [comment]
> Hey @pcaversaccio,
> 
> Took a quick look at the code because I agree it feels weird. So far I agree with the rationale but changing the order of the mapping is a breaking change for upgradeable contracts. I don't think we can fix that until 6.0 and it doesn't harm.

Yeah it doesn't harm and you guys can patch it whenever the timing is ready. I just think it's kinda confusing and should be patched one day at least.

**Comment by ernestognw** [comment]
Sounds good to me, I'm labeling this as a breaking change so we take a look back when we work on a new version.

Thanks for raising it!

**Comment by Morhef** [comment]
.
